src/components/Common/SidebarMenu.tsx

```diff
@@ -0,0 +1,117 @@
+import { useMemo } from 'react';
+
+import { useTranslation } from 'next-i18next';
+
+import classNames from 'classnames';
+
+import { getByHighlightColor } from '@/src/utils/app/folders';
+
+import { BaseMenuItemRendererProps, BaseMenuProps } from '@/src/types/menu';
+
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipTrigger,
+} from '@/src/components/Common/Tooltip';
+
+import BaseContextMenu from './BaseContextMenu';
+
+export function SidebarMenuItemRenderer(props: BaseMenuItemRendererProps) {
+  const {
+    Icon,
+    dataQa,
+    onClick,
+    disabled,
+    highlightColor,
+    translation,
+    className,
+    menuItems,
+  } = props;
+  const item = (
+    <button
+      className={classNames(
+        'flex cursor-pointer items-center justify-center rounded p-[5px] disabled:cursor-not-allowed',
+        getByHighlightColor(
+          highlightColor,
+          'hover:bg-green/15 hover:text-green',
+          'hover:bg-violet/15 hover:text-violet',
+          'hover:bg-blue-500/20 hover:text-blue-500',
+        ),
+        className,
+      )}
+      onClick={!menuItems ? onClick : undefined}
+      data-qa={dataQa}
+      disabled={disabled}
+    >
+      {Icon && <Icon size={24} height={24} width={24} strokeWidth="1.5" />}
+    </button>
+  );
+
+  if (menuItems) {
+    return (
+      <BaseContextMenu
+        menuItems={menuItems}
+        highlightColor={highlightColor}
+        translation={translation}
+        CustomMenuRenderer={item}
+      />
+    );
+  }
+  return item;
+}
+
+export default function SidebarMenu({
+  menuItems,
+  highlightColor,
+  displayMenuItemCount = 5, // calculate in future based on width of container
+  translation = 'chatbar',
```

Comment:
same concern here - I think translations should be resolved on level where you are creating component instance